File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/HeapFile.java
package com.github.ol_loginov.heaplibweb.repository;

import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
public class HeapFile extends EntityIdentity {
	private String relativePath;
	private HeapFileStatus status = HeapFileStatus.PENDING;
	private Instant loadStart = Instant.now();
	private Instant loadFinish;
	private Float loadProgress;
	private String loadMessage = "";
	private String loadError;
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/HeapFile.kt
package com.github.ol_loginov.heaplibweb.repository

import java.time.Instant
import java.time.ZonedDateTime

class HeapFile private constructor(
    id: Int,
    var path: String,
    var status: HeapFileStatus,
    var loadStart: ZonedDateTime,
    var loadFinish: ZonedDateTime?,
    var loadProgress: Float,
    var loadMessage: String,
    var loadError: String?
) : EntityIdentity(id) {
    private constructor() : this(0, "", HeapFileStatus.PENDING, ZonedDateTime.now(), null, 0f, "", null)
    constructor(path: String) : this(0, path, HeapFileStatus.PENDING, ZonedDateTime.now(), null, 0f, "", null)
}
